### PR TITLE
Schedule handles keyword GCONSUMP.

### PR DESCRIPTION
**Pull request by stefoss23**
Keyword GCONSUMP is handled by Schedule.

### CODE REVIEW

**Comment by stefoss23** [comment]
jenkins build this please

File: opm/parser/eclipse/EclipseState/Schedule/Group/GConSump.hpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+#include <map>
+#include <string>
+
+#include <opm/parser/eclipse/Deck/UDAValue.hpp>
+
+namespace Opm {
+
+    class GConSump {
+    public:
+        GConSump();
```
Comment (joakim-hove) [inline opm/parser/eclipse/EclipseState/Schedule/Group/GConSump.hpp]: When you just want the absolute default default constructor you can do:
```
GConSump() = default;
```
and do not need to implement anything in the cpp file.

**Comment by joakim-hove** [comment]
One minor comment about the default constructor - otherwise this is ready to be merged :+1:

File: src/opm/parser/eclipse/EclipseState/Schedule/Group/GConSump.cpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    if (it == groups.end())
+        throw std::invalid_argument("Current GConSump obj. does not contain '" + name + "'.");
+    else
+        return it->second;
+}
+
+
+void GConSump::add(const std::string& name, const UDAValue& consumption_rate, const UDAValue& import_rate, const std::string network_node) {
+    groups[name] = GCONSUMPGroup();
+    GConSump::GCONSUMPGroup& group = groups[name];
```
Comment (bska) [inline src/opm/parser/eclipse/EclipseState/Schedule/Group/GConSump.cpp]: Is `add` supposed to clear out any existing values for `name`?  If not, then the above is wasteful.  The `groups` member is already a `std::map` so the index operator will default construct a `GCONSUMPGroup` object in `name`'s position if one does not already exist.

If you do not need/intend to clear existing values, then these two lines can be simplified to
```C++
GConSump::GConSump& group = this->groups[name];
```

**Comment by stefoss23** [comment]
jenkins build this please

**Pull request merged**